app/connectors/ReferenceDataConnector.scala

```diff
@@ -51,4 +52,18 @@ class ReferenceDataConnector @Inject() (config: FrontendAppConfig, http: HttpCli
           onFailControlType(code)
       }
   }
+
+  def getFunctionalErrorDescription(code: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[FunctionalErrorWithDesc] = {
+    def onFailFunctionalError(code: String): FunctionalErrorWithDesc = FunctionalErrorWithDesc(code, "")
+    val serviceUrl                                                   = s"${config.referenceDataUrl}/functional-error-type/$code"
+
+    http
+      .GET[Option[FunctionalErrorWithDesc]](serviceUrl)
+      .map(_.getOrElse(onFailFunctionalError(code)))
+      .recover {
+        case _ =>
+          logger.error(s"Get Functional Error Type  request failed to return data")
+          onFailFunctionalError(code)
+      }
+  }
```

Comment:
Is this not the same as:

  http
      .GET [FunctionalErrorWithDesc] (serviceUrl)
      .recover {
        case _ =>
          logger.error(s"Get Functional Error Type  request failed to return data")
          onFailFunctionalError(code)
      }
  }

Reply:
when the call returns None, we want to pass a default FunctionalError. so we put getOrElse. :)